sentry/src/main/java/io/sentry/DuplicateEventDetectionEventProcessor.java

```diff
@@ -2,45 +2,63 @@
 
 import io.sentry.util.Objects;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
-import java.util.Map;
-import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
 
 /** Deduplicates events containing throwable that has been already processed. */
 public final class DuplicateEventDetectionEventProcessor implements EventProcessor {
-  private final WeakHashMap<Throwable, Object> capturedObjects = new WeakHashMap<>();
+  private final ConcurrentLinkedQueue<Throwable> capturedObjects = new ConcurrentLinkedQueue<>();
```

Comment:
I'm not confortable with this to be honest. Can we discuss this before going ahead?

Reply:
Sure sure, I was waiting for your blessing anyway :-) We must change the `WeakHashMap` though (https://kdowbecki.github.io/WeakHashMap-is-not-thread-safe/ - this likely contributed to our issues). The option could be to synchronize `WeakHashMap` but this comes with drawbacks too.

Reply:
There's no thread safe `WeakHashMap` in Java standard library, but perhaps we can include this dependency? https://github.com/raphw/weak-lock-free

Reply:
Or: considering that Deduplication is neccessary only with Spring + logging framework integration, perhaps we can move this event processor to Spring module and use `ConcurrentReferenceHashMap` from Spring Framework. This would solve the problem assuming that only thread safety is an issue.

Reply:
We discussed this on a call and decided to go with @maciejwalkowiak's suggestion of using `Collections.synchronizedMap`.